src/org/traccar/WebDataHandler.java

```diff
@@ -121,10 +128,41 @@ public String formatRequest(Position position) {
 
     @Override
     protected Position handlePosition(Position position) {
+        if (json) {
+            AsyncHttpClient.BoundRequestBuilder requestBuilder = Context.getAsyncHttpClient().preparePost(url);
+            requestBuilder.setBodyEncoding(StandardCharsets.UTF_8.name());
 
-        Context.getAsyncHttpClient().prepareGet(formatRequest(position)).execute();
+            requestBuilder.addHeader("Content-Type", "application/json; charset=utf-8");
 
+            requestBuilder.setBody(prepareJsonPayload(position));
+            requestBuilder.execute();
+
+        } else {
+            Context.getAsyncHttpClient().prepareGet(formatRequest(position)).execute();
+        }
         return position;
     }
 
+    protected String prepareJsonPayload(Position position) {
+
+        Map<String, Object> data = new HashMap<>();
+
+        if (position != null) {
+            data.put(KEY_POSITION, position);
+        }
+
+        if (position.getDeviceId() != 0) {
```

Comment:
I think this check if not necessary and next check device for null be enough .

Reply:
@Abyss777, should we remove same checks from `EventForwarder`?

Reply:
@Abyss777  Again, the check of the position is not necessary ? just leave the check device for null ?

Reply:
I think it's safe to assume that there is a device.

Reply:
So remove both checks ?

Reply:
Hm...I believe yes we can remove.
Map.get(0) will return `null`  - "the value to which the specified key is mapped, or null if this map contains no mapping for the key"